truffleruby/src/main/java/org/truffleruby/parser/BodyTranslator.java

```diff
@@ -1685,10 +1683,12 @@ public RubyNode visitGlobalAsgnNode(GlobalAsgnParseNode node) {
 
         switch (name) {
             case "$~":
-                rhs = new CheckMatchVariableTypeNode(rhs);
-                rhs = WrapInThreadLocalNodeGen.create(rhs);
-                rhs.unsafeSetSourceSection(sourceSection);
-                environment.declareVarInMethodScope("$~");
+                environment.declareVarInMethodScope(name);
+                // The right hand side must be built using the resolved variable.
+                break;
+            case "$_":
+                // The right hand side must be built using the resolved variable.
+                environment.declareVar(name);
```

Comment:
Not sure why the original code declared them differently. I've changed them both to declare the variable in the method scope so we are at least consistent.